File: shooting_drill.R
simulate_drill <- function(lower = -5, upper = 7, perc = 0.75){
  score <- 0
  while ((score > lower) && (score < upper)){
    if (runif(1) < perc){
      score <- score + 1
    } else {
      score <- score - 2
    }
  }
  
  if (score == lower){
    result <- 1
  } else {
    result <- 0
  }
  return(result)
}

#TODO make perc a vector so you can output the vector of results with diff shooting percentages


simulate_drills <- function(n = 1, lower = -5, upper = 7, perc = 0.75){
  num_won <- 0
  for (i in 1:n){
    num_won <- num_won + simulate_drill(lower, upper, perc)
  }
  return(num_won)
}

File: shooting_drill.py
# -*- coding: utf-8 -*-
"""
Created on Sun Jan  7 16:12:28 2018

@author: Johnny Wong
"""
import numpy as np

def simulate_drill(lower = -5, upper = 7, perc = 0.7):
	score = 0
	shots = 0
	while (score > lower and score < upper):
		if np.random.rand() < perc:
			score += 1
		else:
			score -= 2
		shots += 1
	
	if score <= lower:
		result = 0
	else:
		result = 1

	return result, shots

'''
def simulate_drills(n = 1, lower = -5, upper = 7, perc = 0.7):
	perc = np.array(list(perc)) #TODO 
	num_won = list(range(perc))
	for i in range(len(perc)):
		for j in range(n):
			num_won[i] += simulate_drill(perc = perc[i])
		
	return num_won
'''